dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java

```diff
@@ -0,0 +1,756 @@
+import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import scala.Predef;
+import scala.Tuple2;
+import scala.collection.JavaConverters;
+import scala.collection.immutable.Map;
+
+import util.TestException;
+
+import static java.lang.Thread.sleep;
+
+public class SnappyTest implements Serializable {
+
+    private static transient JavaSparkContext jsc;
+    private static transient SnappyContext snc;
+    protected static SnappyTest snappyTest;
+    private static char sep;
+    private static HostDescription hd;
+    private static String snappyPath = null;
+    private static String logFile = null;
+    private static String localHost = null;
+
+    private static Set<String> serversFileContent = new LinkedHashSet<String>();
+    private static Set<String> locatorsFileContent = new LinkedHashSet<String>();
+    private static Set<String> leadsFileContent = new LinkedHashSet<String>();
+    private static Set<String> snappyJobLogFilesForTask = new LinkedHashSet<String>();
+    private static Set<String> snappyJobLogFilesForCloseTask = new LinkedHashSet<String>();
+    private static Set<Integer> pids = new LinkedHashSet<Integer>();
+    private static String locatorsFilePath = null;
+    private static String serversFilePath = null;
+    private static String leadsFilePath = null;
+    private static String userAppJar = null;
+    private static String simulateStreamScriptName = TestConfig.tab().stringAt(SnappyPrms.simulateStreamScriptName, "simulateFileStream");
+    private static String simulateStreamScriptDestinationFolder = null;
+    private static String leadHost = null;
+    public static Long waitTimeBeforeJobStatusInTask = TestConfig.tab().longAt(SnappyPrms.jobExecutionTimeInMillisForTask, 6000);
+    public static Long waitTimeBeforeStreamingJobStatusInTask = TestConfig.tab().longAt(SnappyPrms.streamingJobExecutionTimeInMillisForTask, 6000);
+    public static Long waitTimeBeforeJobStatusInCloseTask = TestConfig.tab().longAt(SnappyPrms.jobExecutionTimeInMillisForCloseTask, 6000);
+
+
+    public static <A, B> Map<A, B> toScalaMap(HashMap<A, B> m) {
+        return JavaConverters.mapAsScalaMapConverter(m).asScala().toMap(Predef.<Tuple2<A, B>>conforms());
+    }
+
+    public static void HydraTask_initializeSnappy() {
+        SparkContext sc = null;
+        snc = SnappyContext.getOrCreate(sc);
+        Log.getLogWriter().info("SnappyContext initialized successfully");
+    }
+
+
+    public static void HydraTask_stopSnappy() {
+        snc.stop();
+        Log.getLogWriter().info("SnappyContext stopped successfully");
+    }
+
+    protected String getDataLocation() {
+        String scriptPath = null;
+        try {
+            hd = getHostDescription();
+            scriptPath = hd.getGemFireHome() + sep + ".." + sep + "snappy"
+                    + sep + "quickstart" + sep + "data";
+        } catch (Exception e) {
```

Comment:
what Exception you are expecting here??? why not to throw proper exception or not to catch it at all